src/parser.js

```diff
@@ -1693,14 +1713,12 @@ export class Parser extends Tokenizer {
         return this.markLocation(new Shift.ShorthandProperty(new Shift.Identifier(key.value)), startTokenIndex);
       }
     }
-    if (this.eof() || token.type.klass == TokenClass.Punctuator) {
-      throw this.createUnexpected(token);
-    } else {
-      let key = this.parseObjectPropertyKey();
-      this.expect(TokenType.COLON);
-      let value = this.parseAssignmentExpression();
-      return this.markLocation(new Shift.DataProperty(key, value), startTokenIndex);
-    }
+
+    // PropertyName[?Yield] : AssignmentExpression[In,?Yield]
+    let key = this.parseObjectPropertyKey();
+    this.expect(TokenType.COLON);
```

Comment:
This position allows either a `COLON` for `DataProperty` or a `LPAREN` for `Method`